test/support/test_macros.h

```diff
@@ -94,15 +96,15 @@
 
 /* Make a nice name for the standard version */
 #ifndef TEST_STD_VER
-#if  __cplusplus <= 199711L
+#if  _ONEDPL_CPLUSPLUS <= 199711L
 # define TEST_STD_VER 3
-#elif __cplusplus <= 201103L
+#elif _ONEDPL_CPLUSPLUS <= 201103L
 # define TEST_STD_VER 11
-#elif __cplusplus <= 201402L
+#elif _ONEDPL_CPLUSPLUS <= 201402L
 # define TEST_STD_VER 14
-#elif __cplusplus <= 201703L
+#elif _ONEDPL_CPLUSPLUS <= 201703L
 # define TEST_STD_VER 17
-#elif __cplusplus <= 202002L
+#elif _ONEDPL_CPLUSPLUS <= 202002L
 # define TEST_STD_VER 20
```

Comment:
I think it makes more sense to revert the order, especially when adding the check for `_MSVC_LANG`:

```
#if __cplusplus >= 202002L || _MSVC_LANG >= 202002L
#define TEST_STD_VER 20
#elif __cplusplus >= 201703L || _MSVC_LANG >= 201703L
#define TEST_STD_VER 17
...
```

Reply:
Fixed